src/Analysis/Engine/Impl/Indexing/SymbolIndexWalker.cs

```diff
@@ -149,6 +149,66 @@ internal class SymbolIndexWalker : PythonWalker {
             return false;
         }
 
+        public override bool Walk(ForStatement node) {
+            _stack.EnterDeclared();
+            AddVarSymbolRecursive(node.Left);
+            node.List?.Walk(this);
+            node.Body?.Walk(this);
+            _stack.ExitDeclaredAndMerge();
+
+            _stack.EnterDeclared();
+            node.Else?.Walk(this);
+            _stack.ExitDeclaredAndMerge();
+
+            return false;
+        }
+
+        public override bool Walk(ComprehensionFor node) {
+            AddVarSymbolRecursive(node.Left);
+            return false;
```

Comment:
This should be `base.Walk(node)` to allow nesting to work. I'll add that and a test for it.

Reply:
Fixed.